### PR TITLE
AC-535:Migrate To Androidx

### PR DESCRIPTION
**Pull request by aggarwalpulkit596**
AC-535:Migrate To Androidx

## Description of what I changed
migrated all support libraries to Androidx.

## Issue I worked on
JIRA Issue: https://issues.openmrs.org/browse/AC-535

## Checklist: I completed these to help reviewers :)
- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).
- [x] I have **added tests** to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)
- [x] All new and existing **tests passed**.
- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

**Comment by aggarwalpulkit596** [comment]
@f4ww4z Please Review This PR

**Comment by f4ww4z** [comment]
> @f4ww4z Please Review This PR

@aggarwalpulkit596 No need to say this, I get notified of *any* changes made to existing PRs.

File: gradle.properties [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+org.gradle.jvmargs=-Xmx1536m
+
+# When configured, Gradle will run in incubating parallel mode.
+# This option should only be used with decoupled projects. More details, visit
+# http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
+# org.gradle.parallel=true
+
+android.useAndroidX=true
+android.enableJetifier=true
+android.enableSeparateAnnotationProcessing=true
```
Comment (f4ww4z) [inline gradle.properties]: Why is this setting useful?

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginContract.java [hunk 2]
```diff
@@ -23,10 +25,15 @@
...
 
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
 public interface LoginContract {
 
     interface View extends BaseView<Presenter>{
 
+        android.view.View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState);
+
```
Comment (f4ww4z) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginContract.java]: Why is this added?

File: gradle.properties (hunk 1 above)
Comment (aggarwalpulkit596) [reply gradle.properties]: For the use of androidx we need to enable this and android.enableSeparateAnnotationProcessing=true is required for annotation processor library otherwise travis CI would fail

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginContract.java (hunk 2 above)
Comment (aggarwalpulkit596) [reply openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginContract.java]: It might have been accidentally added would remove it right away

Comment (f4ww4z) [reply openmrs-client/src/main/java/org/openmrs/mobile/activities/login/LoginContract.java]: @aggarwalpulkit596 Yes, I've finished reviewing. So please remove this (line 35) as `onCreateView` wasn't created before. Do also remove extra lines.

File: gradle.properties (hunk 1 above)
Comment (f4ww4z) [reply gradle.properties]: @aggarwalpulkit596 O.K., looks good.

**Comment by aggarwalpulkit596** [comment]
@f4ww4z Done
